akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerSpec.scala

```diff
@@ -648,6 +648,121 @@ class HttpServerSpec extends AkkaSpec(
       netOut.expectComplete()
     })
 
+    "produce a `100 Continue` response when requested by a `Default` entity and some data sent early" in assertAllStagesStopped(new TestSetup {
+      send("""POST / HTTP/1.1
+             |Host: example.com
+             |Expect: 100-continue
+             |Content-Length: 24
+             |
+             |0123456789ABCDEF""") // only 16 bytes get send at first
+      inside(expectRequest()) {
+        case HttpRequest(POST, _, _, Default(ContentType(`application/octet-stream`, None), 24, data), _) =>
+          val dataProbe = TestSubscriber.manualProbe[ByteString]()
+          data.to(Sink.fromSubscriber(dataProbe)).run()
+          val dataSub = dataProbe.expectSubscription()
+          netOut.expectNoBytes(50.millis.dilated)
+          dataSub.request(1) // triggers `100 Continue` response
+          expectResponseWithWipedDate(
+            """HTTP/1.1 100 Continue
```

Comment:
Ok, a server is not required to send a `100 Continue` response once it has seen some data but let's not change it right now.